src/ui/public/index_patterns/_index_pattern.js

```diff
@@ -293,8 +293,19 @@ define(function (require) {
       };
 
       self._fetchFields = function () {
+        var existingFieldsByName = self.fields.byName;
+
         return mapper.getFieldsForIndexPattern(self, true)
         .then(function (fields) {
+
+          // copy over kibana-added properties from existing fields
+          fields.forEach(function (field) {
```

Comment:
Just for my own knowledge. Why do we need to treat the `exclude` attribute in a special way ? Other facts like `doc_values` are not.